frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java

```diff
@@ -87,66 +85,39 @@ public void actionPerformed(AnActionEvent anActionEvent) {
       }
       val currentBranchName = Option.of(gitRepository.getCurrentBranch()).map(b -> b.getName()).getOrNull();
 
-      if (FetchAllRemotesAction.isUpToDate(gitRepository)) {
-        val ffmProps = new FastForwardMergeProps(
-            /* movingBranch */ localBranch,
-            /* stayingBranch */ remoteBranch);
+      val ffmProps = new FastForwardMergeProps(
+          /* movingBranch */ localBranch,
+          /* stayingBranch */ remoteBranch);
 
-        if (localBranchName.equals(currentBranchName)) {
-          BaseFastForwardMergeBranchToParentAction.doFastForwardCurrentBranch(project, gitRepository, ffmProps);
-        } else {
-          BaseFastForwardMergeBranchToParentAction.doFastForwardNonCurrentBranch(project, gitRepository, ffmProps);
-        }
+      Runnable fastForwardRunnable = () -> {};
+      if (localBranchName.equals(currentBranchName)) {
+        fastForwardRunnable = () -> BaseFastForwardMergeBranchToParentAction.doFastForwardCurrentBranch(project, gitRepository,
+            ffmProps);
+      } else {
+        fastForwardRunnable = () -> BaseFastForwardMergeBranchToParentAction.doFastForwardNonCurrentBranch(project,
+            gitRepository, ffmProps);
+      }
 
+      if (FetchAllRemotesAction.isUpToDate(gitRepository)) {
+        fastForwardRunnable.run();
       } else {
-        if (localBranchName.equals(currentBranchName)) {
-          doPullCurrentBranchFastForwardOnly(project, gitRepository, remoteBranch);
-        } else {
-          doPullNonCurrentBranchFastForwardOnly(project, gitRepository, localBranch, remoteBranch);
-        }
+        updateRepositoryBackgroundable(project, gitRepository, remoteBranch, /* onSuccessRunnable */ fastForwardRunnable);
       }
     }
   }
 
-  private void doPullCurrentBranchFastForwardOnly(Project project,
+  private void updateRepositoryBackgroundable(Project project,
```

Comment:
Hmm maybe let's refer to `fetch` from the name of this method, to distinguish it from `fastForwardRunnable` from the other case